web/src/components/storage/ProposalSpacePolicySection.jsx

```diff
@@ -0,0 +1,457 @@
+      <Td dataLabel={columnNames.size} textCenter><DeviceSizeColumn device={device} /></Td>
+      <Td dataLabel={columnNames.action} textCenter>
+        <If
+          condition={!isDrive(device)}
+          then={
+            <DeviceActionColumn
+              device={device}
+              action={spaceAction?.action || "keep"}
+              isDisabled={isDisabled}
+              onChange={onChange}
+            />
+          }
+        />
+      </Td>
+    </TreeRowWrapper>
+  );
+};
+
+/**
+ * Table for configuring the space actions.
+ * @component
+ *
+ * @param {object} props
+ * @param {ProposalSettings} props.settings
+ * @param {(action: SpaceAction) => void} [props.onChange]
+ */
+const SpaceActionsTable = ({ settings, onChange = noop }) => {
+  const [expandedDevices, setExpandedDevices] = useLocalStorage("storage-space-actions-expanded", []);
+  const [autoExpanded, setAutoExpanded] = useLocalStorage("storage-space-actions-auto-expanded", false);
+
+  useEffect(() => {
+    const policy = settings.spacePolicy;
+    const devices = settings.installationDevices.map(d => d.name);
+    let currentExpanded = devices.filter(d => expandedDevices.includes(d));
+
+    if (policy === "custom" && !autoExpanded) {
+      currentExpanded = [...devices];
+      setAutoExpanded(true);
+    } else if (policy !== "custom" && autoExpanded) {
+      setAutoExpanded(false);
+    }
+
+    if (currentExpanded.sort().toString() !== expandedDevices.sort().toString()) {
+      setExpandedDevices(currentExpanded);
+    }
+  }, [autoExpanded, expandedDevices, setAutoExpanded, setExpandedDevices, settings]);
+
+  const renderRows = () => {
+    const rows = [];
+
+    settings.installationDevices?.forEach((device, index) => {
+      const isExpanded = expandedDevices.includes(device.name);
+      const children = device.partitionTable?.partitions;
+
+      const onCollapse = () => {
+        const otherExpandedDevices = expandedDevices.filter(name => name !== device.name);
+        const expanded = isExpanded ? otherExpandedDevices : [...otherExpandedDevices, device.name];
+        setExpandedDevices(expanded);
+      };
+
+      rows.push(
+        <DeviceRow
+          key={device.name}
+          device={device}
+          settings={settings}
+          rowIndex={rows.length}
+          setSize={children?.length || 0}
+          posInSet={index}
+          isExpanded={isExpanded}
+          onCollapse={onCollapse}
+          onChange={onChange}
+        />
+      );
+
+      children?.forEach((child, index) => {
+        rows.push(
+          <DeviceRow
+            key={child.name}
+            device={child}
+            settings={settings}
+            rowIndex={rows.length}
+            level={2}
+            posInSet={index}
+            isHidden={!isExpanded}
+            onCollapse={onCollapse}
+            onChange={onChange}
+          />
+        );
+      });
+    });
+
+    return rows;
+  };
```

Comment:
Note for another iteration: as we were talking, most probably this deserves a core/TreeTable wrapper or so. Specially given the fact we are adding an extra feature to the component provided by the library (the state saved into localStorage).